### ISSUE TITLE
apython exits when there is too much output

### ISSUE THREAD

**Issue by gnachman** [vxgmichel/aioconsole#42]
To reproduce:

1. Run `apython`
2. Do something that produces a lot of output, such as `help(str)`

The apython shell terminates abruptly, exiting and returning to the shell.

With a bit of hacking around I got a stack trace out of it:

```
  File "/usr/local/bin/apython", line 11, in <module>
    sys.exit(run_apython())
  File "/usr/local/lib/python3.7/site-packages/aioconsole/apython.py", line 97, in run_apython
    code = run_apython_in_subprocess(args, namespace.prompt_control)
  File "/usr/local/lib/python3.7/site-packages/aioconsole/apython.py", line 154, in run_apython_in_subprocess
    use_stderr=True)
  File "/usr/local/lib/python3.7/site-packages/aioconsole/rlwrap.py", line 23, in rlwrap_process
    return _rlwrap(process, prompt_control, use_stderr)
  File "/usr/local/lib/python3.7/site-packages/aioconsole/rlwrap.py", line 44, in _rlwrap
    prompt = future.result()
  File "/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/lib/python3.7/concurrent/futures/_base.py", line 432, in result
    return self.__get_result()
  File "/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/lib/python3.7/concurrent/futures/_base.py", line 384, in __get_result
    raise self._exception
  File "/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.7/site-packages/aioconsole/rlwrap.py", line 91, in wait_for_prompt
    write(current)
  File "/usr/local/lib/python3.7/site-packages/aioconsole/rlwrap.py", line 84, in write
    dest.flush()
BlockingIOError: [Errno 35] write could not complete without blocking
```

It can be fixed (apparently) by adding this to the top of `wait_for_prompt`:

```
fcntl.fcntl(dest.fileno(), fcntl.F_SETFL, 0)
```

For some reason removing the O_NONBLOCK flag has to be done the *second* time `wait_for_prompt` is called. I can't figure out where it's getting changed.

**Comment by vxgmichel** [comment]
Thanks for the report @gnachman !

Interesting, I can't reproduce the issue even though I'm using a similar setup:
- python 3.7.0 on linux
- aioconsole 0.1.11
- `help(str)` and `print('a'*10**7)` both run fine

Any idea what the difference could be?

Also, do you get the same behavior without readline support (`apython --no-readline`)?

**Comment by gnachman** [comment]
I'm on macOS, aconsole 0.1.11. `--no-readline` prevents the issue from happening.

I wouldn't be surprised if readline is doing something weird or insane. I've seen it before.

**Comment by gnachman** [comment]
I've sent PR 43 to implement the workaround. I'd like to ship aioconsole with iTerm2 in the near future but this bug is a blocker.

**Comment by vxgmichel** [comment]
Thanks a lot for the fix, it's now available with release [v0.1.12](https://github.com/vxgmichel/aioconsole/releases/tag/v0.1.12), already deployed on pypi :smiley:
